cpp/src/arrow/util/binary_view_util.h

```diff
@@ -34,22 +35,27 @@ inline BinaryViewType::c_type ToInlineBinaryView(const void* data, int32_t size)
 }
 
 inline BinaryViewType::c_type ToInlineBinaryView(std::string_view v) {
+  assert(v.size() <= BinaryViewType::kInlineSize);
   return ToInlineBinaryView(v.data(), static_cast<int32_t>(v.size()));
 }
 
-inline BinaryViewType::c_type ToBinaryView(const void* data, int32_t size,
-                                           int32_t buffer_index, int32_t offset) {
-  if (size <= BinaryViewType::kInlineSize) {
-    return ToInlineBinaryView(data, size);
-  }
-
+inline BinaryViewType::c_type ToLargeBinaryView(const void* data, int32_t size,
```

Comment:
nit: call this `ToNonInlineBinaryView` to not confuse people that might think a `LARGE_BINARY_VIEW` type exists. [I misread it like this at first :)]